### PR TITLE
Make TestLauncher be able to run tasks along with tests

### PR DESCRIPTION
**Pull request by donat**
image

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by donat** [comment]
[](https://github.com/gradle/gradle/pull/20073#issuecomment-1060876189)@bot-gradle test RFM

**Comment by donat** [comment]
@bot-gradle test RFM

File: subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/action/BuildActionSerializer.java [hunk 1]
```diff
@@ -342,13 +342,17 @@ public ClientProvidedPhasedAction read(Decoder decoder) throws Exception {
...
 
-        public TestExecutionRequestPayload(Set<InternalTestDescriptor> testDescriptors, Set<String> classNames, Set<InternalJvmTestRequest> internalJvmTestRequests, InternalDebugOptions debugOptions, Map<String, List<InternalJvmTestRequest>> taskAndTests) {
+        public TestExecutionRequestPayload(Set<InternalTestDescriptor> testDescriptors, Set<String> classNames, Set<InternalJvmTestRequest> internalJvmTestRequests, InternalDebugOptions debugOptions, Map<String, List<InternalJvmTestRequest>> taskAndTests, boolean isRunDefaultTasks, List<String> tasks) {
             this.testDescriptors = testDescriptors;
             this.classNames = classNames;
             this.internalJvmTestRequests = internalJvmTestRequests;
             this.debugOptions = debugOptions;
             this.taskAndTests = taskAndTests;
+            this.isRunDefaultTasks = isRunDefaultTasks;
+            this.tasks = tasks;
```
Comment (donat) [inline subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/action/BuildActionSerializer.java:355]: @sterling Maybe we should encapsulate these fields into a new `InternalTaskRequest` class. WDYT?

**Comment by donat** [comment]
@bot-gradle test RFM

**Comment by donat** [comment]
@bot-gradle test rfm

**Comment by donat** [comment]
@bot-gradle test rfm

**Review by big-guy — APPROVED** [review]

**Comment by donat** [comment]
@bot-gradle test and merge

**Comment by donat** [comment]
@bot-gradle cancel

**Comment by donat** [comment]
@bot-gradle test and merge

**Comment by donat** [comment]
@bot-gradle test and merge

**Pull request merged**